File: Coding Exercises/TypeScript/Day13/separateZeroAndOne.ts
function separateZeroAndOne(arr: number[]): void {
  let countZeros = 0;

  for (let i = 0; i < arr.length; i++) {
      if (arr[i] === 0) {
          countZeros++;
      }
  }

  for (let i = 0; i < countZeros; i++) {
      arr[i] = 0;
  }

  for (let i = countZeros; i < arr.length; i++) {
      arr[i] = 1;
  }
}

File: Coding Exercises/Python/Day13/separate_zero_and_one.py
def separate_zero_and_one(arr):
    count_zeros = 0

    for i in range(len(arr)):
        if arr[i] == 0:
            count_zeros += 1

    for i in range(count_zeros):
        arr[i] = 0

    for i in range(count_zeros, len(arr)):
        arr[i] = 1
